File: app/src/main/java/com/codemybrainsout/imageviewer/view/collection/CollectionModule.java
package com.codemybrainsout.imageviewer.view.collection;

import android.content.Context;

import com.codemybrainsout.imageviewer.api.CollectionService;
import com.codemybrainsout.imageviewer.dagger.scope.ActivityScope;

import dagger.Module;
import dagger.Provides;

/**
 * Created by ahulr on 10-06-2017.
 */

@Module
public class CollectionModule {

    private Context context;

    public CollectionModule(Context context) {
        this.context = context;
    }

    @ActivityScope
    @Provides
    Context context() {
        return context;
    }

    @ActivityScope
    @Provides
    CollectionAdapter collectionAdapter(Context context) {
        return new CollectionAdapter(context);
    }

    @ActivityScope
    @Provides
    CollectionPresenter collectionPresenter(CollectionService collectionService) {
        return new CollectionPresenter(collectionService, (CollectionContract.View) context);
    }

}

File: app/src/main/java/com/codemybrainsout/imageviewer/view/collection/CollectionModule.kt
package com.codemybrainsout.imageviewer.view.collection

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

/**
 * Created by ahulr on 10-06-2017.
 */
@Module
@InstallIn(ActivityComponent::class)
abstract class CollectionModule {

    @Binds
    abstract fun bindActivity(activity: CollectionActivity): CollectionContract.View

    @Binds
    abstract fun bindPresenter(impl: CollectionPresenter): CollectionContract.Presenter

}